android/src/main/java/com/margelo/nitro/nitrosse/NitroSse.kt

```
@@ -490,34 +462,108 @@ class NitroSse : HybridNitroSseSpec(), DefaultLifecycleObserver {
+                        val bytesRead = super.read(sink, byteCount)
+                        if (bytesRead != -1L) {
+                            totalBytesReceived.addAndGet(bytesRead)
+                            try {
+                                for (i in 0 until bytesRead) {
+                                    val b = sink.get(bufferOffset + i)
+                                    if (isAtStartOfLine && b == ':'.code.toByte()) {
+                                        onHeartbeat()
+                                    }
+                                    isAtStartOfLine = (b == '\n'.code.toByte() || b == '\r'.code.toByte())
+                                }
+                            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)